rust_snuba/src/strategies/processor.rs

```diff
@@ -57,18 +130,24 @@ pub fn get_schema(schema_name: &str, enforce_schema: bool) -> Option<Arc<Schema>
 }
 
 #[derive(Clone)]
-struct MessageProcessor {
+struct MessageProcessor<TResult: Clone, TNext: Clone> {
     schema: Option<Arc<Schema>>,
     enforce_schema: bool,
-    func: ProcessingFunction,
+    // Convert payload to either InsertBatch (or either insert or replacement for the errors dataset)
+    func:
+        fn(KafkaPayload, KafkaMessageMetadata, config: &ProcessorConfig) -> anyhow::Result<TResult>,
```

Comment:
why'd we get rid of the type alias here?

Reply:
I changed it to a generic, as this can be either `ProcessingFunction` or `ProcessingFunctionWithReplacements` now